source/fundamentals/poco.txt

```diff
@@ -0,0 +1,249 @@
+.. _csharp-poco:
+
+===============
+Work with POCOs
+===============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about how you can use "Plain Old CLR/Class
+Objects", or **POCOs**, with the {+driver-short+} for your operations
+and queries. POCOs are simple class objects which do not inherit
+features from any framework-specific base class and do not return any
+special types. We recommend using POCOs in your {+language+} code to
+adhere to idiomatic driver usage and achieve the best performance.
+
+You should read this guide if you want to learn more about how to use
+POCOs with the {+driver-short+} or need to adjust the driver's default
+field mapping behavior.
+
+Create a POCO
+-------------
+
+You can create a POCO by instantiating any simple class that does not
+implement interfaces or extend classes from a framework. When you
+execute an operation such as a read or write using a POCO, the driver
+interally *serializes*, or converts, the POCO to BSON.
+
+Select the :guilabel:`POCO` or :guilabel:`BSON` tab to see how the
+driver serializes a sample POCO to BSON:
+
+.. tabs::
+
+   .. tab:: POCO
+      :tabid: poco-representation
+
+      .. code-block:: csharp
+         :copyable: false
+
+         public class Clothing
+         {
+             public ObjectId _id { get; set; }
+             public string name { get; set; }
+             public bool instock { get; set; }
+             public double price { get; set; }
+             public List<string> color { get; set; }
+         }
+
+   .. tab:: BSON
+      :tabid: bson-representation
+
+      .. code-block:: json
+         :copyable: false
+
+         {
+           "_id": ObjectId("..."),
+           "name": "Long Sleeve Shirt",
+           "instock": true,
+           "price": 17.99,
+           "color": [ "black", "navy", "red" ]
+         }
+
+You can define a POCO with any object structure that suits your
+needs, including nested objects, arrays, lists, and any data types. The
+driver serializes POCOs using default mapping rules to generate the
+appropriate BSON representation of the document.
+
+Custom Serialization
+--------------------
+
+If the default field mapping behavior does not meet your needs, you can
+specify custom behavior using serialization related attributes. There
+are various attributes which you can specify to modify the way that the
+driver serializes each property of your POCO. This section describes
+some of the common serialization related attributes.
+
+Set Field Names
```

Comment:
Issue: `BsonElement` works for setting field names, but it onerous if there are lots of these fields.

Suggestion: In a tip box, mention that if all you want to do is map PascalCase properties to camelCase fields, you can use code like this in the program instead of `BsonElement`:

```
// This allows automapping of the camelCase database fields to our models. 
var camelCaseConvention = new ConventionPack { new CamelCaseElementNameConvention() };
ConventionRegistry.Register("CamelCase", camelCaseConvention, type => true);
```